### ISSUE TITLE
Bug in Persist State Similar to #19

### ISSUE THREAD

**Issue by thk-socal** [Anamico/node-red-contrib-alarm#20]
Commented out SecuritySystemTargetState setting in panel.js near the "// persist the new state" section causes the same hang up as bug #19. Causes Homekit to hang on state change. I edited the code as below for a quick fix for my situation, but I think there needs to be some checking on my change.

![image](https://user-images.githubusercontent.com/22800990/109916386-6356ad80-7c68-11eb-9b58-809e7094cea9.png)

**Comment by cscott365** [comment]
The more I think through this...I should be able to send a SecuritySystemTargetState command to the Alarm panel at the start of a process, and then send the SecurityStateCurrentState at the end of the process. That way I get the "Arming..." interface in Homekit and if I interrupt the process, like for an open door, the Arming process is cancelled and returned to the previous state without success.

**Comment by cscott365** [comment]
Reviewing the code, looks like there needs to be a global variable created for the SecuritySystemTargetState and then logic adjusted around that. The code fixes just assign it to the CurrentState, which is not the intent of the code, but gets around the error for now. This is more than a couple of lines of fixes and still learning all the areas of the code before recommending a change. Might have time to tackle some of it this weekend if you do not get to it before then. Pretty nice node-red module for a custom alarm, just a few bugs to work through to really get it solid in Homekit!

**Comment by macinspak** [comment]
Thanks. And thanks for looking into this enhancement.

The original intention was pretty simple, just return the new state based
on the request. It met the main need to have the alarm arm and disarm
instantly.

With some enhancement, then the behavior you suggested sounds like it could
be useful. Would just need to make sure we can do it without causing
unintended side effects like timeouts or state confusion in fast switching.
But what you suggested sounds great to me. If you want to do that in a fork
and submit a pull request we can get the library even better.

Regards,
Andrew

**Comment by cscott365** [comment]
Trying to stay with the original intent then, which I completely agree with, I did the work in node-red and just used the modified code above to make sure that this alarm code was sending both states. My node-red flow looks like this now:

![image](https://user-images.githubusercontent.com/22800990/110022815-43170500-7ce1-11eb-98eb-1d24dad8efe6.png)

So after setting the SecuritySystemTargetState setting with the recommended changes above, I was able to get everything to arm and disarm without causing errors. If you just want to look at making that change cool, if you want me to fork and then submit a pull for just that change let me know. 

For future, I will start to consider how the panel might have more inputs, but also how not to complicate it and make it too hard for others.
